webplugin/js/app/components/typing-service.js

```diff
@@ -0,0 +1,100 @@
+function TypingService(appOptions) {
```

Comment:
it is a typing indicator service
also see if we should use class instead of a constructor function

Reply:
will change this in class when es6 support pr will be merge